### ISSUE TITLE
Allow local direct streaming

### ISSUE THREAD

**Issue by Mrc527** [wnielson/Plex-Remote-Transcoder#28]
Hello, I'm proposing you this new feature.
Basically I would like to have an option to say only transcoding should be forwarded to the remote machine. Direct Streaming can be easily handled by the host, without heavy load.

My current environment is the following:
I have a VM which is hosting the PMS. The VM runs over a couple of i3 processors. They're powerful enough even for 1080p transcoding, but they cannot handle multiple requests at the same time.
Then I have a remote machine, powerful enough to handle10 transcoding sessions but as I wrote: it's a remote one. Therefore I'm consuming internet bandwidth to make it work. That's actually not an issue right now but it's useless to waste such an high amount of bandwidth just to change the container of the content.

Long story short: Direct streaming can he easily handled by my i3 host machine, transcoding will be forwarded to the remote one.

What do you think? Is it worth to be implemented?

Bye,
Marco.

**Comment by wnielson** [comment]
Thanks for the feedback.  Yes, I think it is both possible and a good idea to be able to identify the request as either a transcode, direct play or direct stream and perform an appropriate action.  Direct play is already 'supported' in the sense that `prt` doesn't even get involved because the transcoder is not used.  I suspect it is relatively trivial to identify a direct stream but we'll have to investigate.

**Comment by Mrc527** [comment]
It would be great to have it. If it's easy...that will help me a lot :)

I know Direct Play has nothing to do with PRT, that's already a good starting point. But unfortunately, web play is more or less always requiring Direct Streaming. :(

**Comment by psyciknz** [comment]
I know it's a couple of years on, but any ideas how this would get identified?  I'm seeing with 1.16 direct stream videos being pushed to my slave, and becoming a little stuttery.  So forcing them local would be preferable.
